src/commands/list-kinesis-streams.js

```diff
@@ -0,0 +1,227 @@
+}
+
+ListKinesisStreamsCommand.description = "Lists the Kinesis streams";
+ListKinesisStreamsCommand.flags = {
+	region: flags.string({
+		char: "r",
+		description: "AWS region, e.g. us-east-1",
+		required: false
+	}),
+	profile: flags.string({
+		char: "p",
+		description: "AWS CLI profile name",
+		required: false
+	})
+};
+
+const getStreamsDescriptionFromRegion = async region => {
+	const Kinesis = new AWS.Kinesis({ region });
+	let streamDetails = await Kinesis.listStreams({ Limit: 100 }).promise();
+	let streamNames = streamDetails.StreamNames;
+	while (streamDetails.HasMoreStreams) {
+		streamDetails = await Kinesis.listStreams({
+			Limit: 100,
+			ExclusiveStartStreamName: streamDetails.slice(-1)[0]
+		}).promise();
+		streamNames = streamDetails.StreamNames;
+	}
+	const usageStats = await getUsageMetrics(streamNames, region);
+	return describeStreams(streamNames, region, usageStats);
+};
+
+const getStreamsDescriptionFromAllRegions = async () => {
+	const promises = Lambda.regions.map(region =>
+		getStreamsDescriptionFromRegion(region)
+	);
+	const results = await Promise.all(promises);
+	return _.flatMap(results);
+};
+
+const describeStreams = async (streams, region, usageStats) => {
+	const streamsDescription = [];
+	for (const stream of streams) {
+		streamsDescription.push(await describeStream(stream, region, usageStats[stream]));
+	}
+
+	return streamsDescription;
+};
+
+const describeStream = async (streamName, region, streamStat) => {
+	const Kinesis = new AWS.Kinesis({ region });
+	const resp = await Kinesis.describeStream({
+		StreamName: streamName
+	}).promise();
+
+	return {
+		streamName: streamName,
+		arn: resp.StreamDescription.StreamARN,
+		status: resp.StreamDescription.StreamStatus,
+		shards: resp.StreamDescription.Shards,
+		readMbPercentage: formatFloat(
+			(streamStat["GetRecords.BytesSum"] /
+				(resp.StreamDescription.Shards.length *
+					BYTES_OUTGOING_PER_SHARD *
+					ONE_MIN_IN_SECONDS)) *
+				100
+		),
+		readRecordPercentage: formatFloat(
+			(streamStat["GetRecords.RecordsSum"] /
+				(resp.StreamDescription.Shards.length *
+					MAX_RECORDS_PER_SHARD *
+					ONE_MIN_IN_SECONDS)) *
+				100
+		),
+		writeMbPercentage: formatFloat(
+			(streamStat.IncomingBytesSum /
+				(resp.StreamDescription.Shards.length *
+					BYTES_INCOMING_PER_SHARD *
+					ONE_MIN_IN_SECONDS)) *
+				100
+		),
+		writeRecordPercentage: formatFloat(
+			(streamStat.IncomingRecordsSum /
+				(resp.StreamDescription.Shards.length *
+					MAX_RECORDS_PER_SHARD *
+					ONE_MIN_IN_SECONDS)) *
+				100
+		)
+	};
+};
+
+const getUsageMetrics = async (streamNames, region) => {
+	const CloudWatch = new AWS.CloudWatch({ region });
+	const metricNames = [
+		["IncomingBytes", "Sum"],
+		["IncomingRecords", "Sum"],
+		["GetRecords.Bytes", "Sum"],
+		["GetRecords.Records", "Sum"]
+	];
+
+	const oneMinuteAgo = new Date();
+	oneMinuteAgo.setMinutes(oneMinuteAgo.getMinutes() - 1);
+
+	const queries = _.flatMap(streamNames, streamName =>
```

Comment:
you'd need to chunk this, CW only allow you to run X number of queries per request

Reply:
nvm, just saw you're doing it later